src/Gui/TaskView/TaskDialog.cpp

```diff
@@ -96,6 +103,21 @@ bool TaskDialog::canClose() const
     return (ret == QMessageBox::Yes);
 }
 
+void TaskDialog::associateToObject3dView(App::DocumentObject* obj)
+{
+    if (!obj) {
+        return;
+    }
+
+    Gui::Document* guiDoc = Gui::Application::Instance->getDocument(obj->getDocument());
+    auto views = guiDoc->getMDIViewsOfType(View3DInventor::getClassTypeId());
+    if (!views.empty()) {
+        auto* iview = dynamic_cast<View3DInventor*>(views.front());

```

Comment:
This is very problematic because there is no guarantee that the 1st of all 3d views of a document is used for editing.

Especially when editing a sketch this leads to problems if the document has two 3d views and the 2nd is active then this will be used for editing. But if the user closes the 1st 3d view the task dialog will be closed, too.

Another problem of this method is that it's too focused on 3d views (OK it's say in its method name). But what about other view types, e.g. those from TechDraw? Should then there be similar methods for every view type?

This whole method should be removed.

Reply:
Indeed it's not ideal if user has several 3dviews. But I don't know if so many users are using multiple 3dviews (I couldn't even find how to do this when I tried earlier).

Having said that we can still fix this perhaps. What about :
- closing the dialog only if it's the last 3dview being closed then?

This method is a helper method for the case of 3dviews. I don't see the problem with this. 90% of tools are going to do associate with 3dview if any, a helper makes sense. If other kind of views require a helper then it can be done later.

Reply:
> But I don't know if so many users are using multiple 3dviews (I couldn't even find how to do this when I tried earlier).

The number of users is quite irrelevant. Because it's a useful option to have a 2nd view it must be made sure not to cause any strange behaviour. 

> closing the dialog only if it's the last 3dview being closed then?

No, this won't work either -- at least not in general because an edit function (like sketcher) may make special settings to only one of the 3d views.

The implementation of this method is just too inflexible to cover a wide range of use cases.

Reply:
> No, this won't work either -- at least not in general because an edit function (like sketcher) may make special settings to only one of the 3d views.

Well then it makes perfect sense to attach to a single 3d view then. Just not the first but to the one that is currently active. Does that sound correct to you?

If no please offer a solution to this that is ok for you.